server/src/main/java/org/elasticsearch/search/searchafter/SearchAfterBuilder.java

```diff
@@ -199,7 +199,7 @@ private static Object convertValueFromSortType(String fieldName, SortField.Type
 
                 case STRING_VAL:
                 case STRING:
-                    return format.parseBytesRef(value.toString());
+                    return format.parseBytesRef(value);
```

Comment:
Do we have to add a `SortField.Type.BINARY` for this? Is `SortField.Type.CUSTOM` a better option here?